### PR TITLE
Handle toolbar in small sizes with an overflow button

### PR DESCRIPTION
**Pull request by ckaiser**
Fixes the toolbar side of #4708

### CODE REVIEW

File: src/app/ui/toolbar.cpp [hunk 1]
```diff
@@ -616,6 +675,95 @@ void ToolBar::drawToolIcon(Graphics* g, int group_index, SkinPartPtr skin, os::S
...
+
+    ToolGroup* tool_group = *it;
+    for (ToolIterator bit = toolbox->begin(); bit != toolbox->end(); ++bit) {
+      Tool* tool = *bit;
+      if (tool->getGroup() == tool_group) {
+        auto item = new ButtonSet::Item();
+        // TODO: Tooltip with tool->getText(), implement in ButtonSet?
+        item->setIcon(theme->getToolPart(tool->getId().c_str()));
+        item->setId(tool->getId().c_str());
+        item->setMaxSize(toolIconSize);  // TODO: Need a better way to enforce the button/icon size, this one doesn't look great
```
Comment (ckaiser) [inline src/app/ui/toolbar.cpp:723]: @dacap Is there a way to constrain the icon size here without changing how ButtonSet renders things?
image

It doesn't look super great with some of the icons, but the default size without specifying anything is massive:
image

**Comment by dacap** [comment]
One thing I didn't mention about it, is that we should try to avoid merging clang-format changes at the moment. The #4654 was not yet merged because I have some branches modifying several files (including the toolbar to support a left-side toolbar :cold_sweat:).

**Comment by ckaiser** [comment]
Ah, yeah - I wanted to run it mostly for clang-tidy but it formatted everything and I didn't have it committed beforehand 😭. Most of the changes in the header are indents so I'll just revert them

**Comment by dacap** [comment]
The grouped popup at the bottom is smaller than the regular group used in other tool groups:

image

It should be of the same size that a regular tool group.

**Comment by dacap** [comment]
Still the buttons are misaligned (compare the rectangle/ellipses as a regular tool group and as items of the last hidden groups):

image

**Comment by dacap** [comment]
I've just merged 9b90159d1bdd019e24ef41f6fc76650843d99c2c with some extra patches in a600c0012727d61827b6131ba3bb822313ccfb18 (some details in its commit description):

https://github.com/user-attachments/assets/b78550f7-650a-4934-bbe8-eec0d8539321

**Pull request closed without merging**